### PR TITLE
Fix falsy attributes being ignored

### PR DESCRIPTION
**Pull request by jacobp100**

### CODE REVIEW

**Comment by Aweary** [comment]
LGTM

File: test/ReactWrapper-spec.js [hunk 1]
```diff
@@ -429,15 +429,24 @@ describeWithDOM('mount', () => {
...
+      expect(wrapper.find('[value=1]')).to.have.length(1);
+      expect(wrapper.find('[value=2]')).to.have.length(0);
+
+      expect(wrapper.find('[value=false]')).to.have.length(1);
+      expect(wrapper.find('[value=true]')).to.have.length(1);
+
+      expect(wrapper.find('[value="0"]')).to.have.length(0);
+      expect(wrapper.find('[value="1"]')).to.have.length(0);
+      expect(wrapper.find('[value="false"]')).to.have.length(0);
+      expect(wrapper.find('[value="true"]')).to.have.length(0);
```
Comment (ljharb) [inline test/ReactWrapper-spec.js]: Can we add tests for `wrapper.find('[value]')`, and also include `<a value={undefined} />` and `<a value={null} />`? (in both shallow and mount)

Comment (Aweary) [inline test/ReactWrapper-spec.js]: @ljharb we do already have tests for value-free attribute queries [here](https://github.com/jacobp100/enzyme/blob/35bb46f608702f0ae0fe0f0dc63d5b3d9d06a5a4/test/ReactWrapper-spec.js#L568)

Comment (ljharb) [inline test/ReactWrapper-spec.js]: Gtk, but I think it would still be useful in this test case, among a number of elements with falsy-valued props

Comment (jacobp100) [inline test/ReactWrapper-spec.js]: There's tests for null and undefined in `should support object property selectors`.

Comment (ljharb) [inline test/ReactWrapper-spec.js]: I specifically want to check for string selectors, since that's what's affected by this PR.

**Comment by Aweary** [comment]
@jacobp100 can you add `<a value={undefined} />` and `<a value={null} />` to the shallow and mount test and resolve the merge conflicts? I'd like to merge this afterwards 👍

**Comment by lencioni** [comment]
@jacobp100 when you have a moment, would you also mind rebasing this onto latest master in addition to adding the tests requested by https://github.com/airbnb/enzyme/pull/459#issuecomment-239576011?

**Comment by ljharb** [comment]
This seems like it will be superseded by #662.

**Pull request closed without merging**